cosmos_rl/rollout/trtllm_rollout/trtllm_worker.py

```diff
@@ -0,0 +1,628 @@
+                nccl_unique_id_key
+            )
+            if nccl_group_id is None:
+                raise RuntimeError(
+                    "[Rollout] Failed to query nccl group_id from controller!"
+                )
+            # create the communicator index
+            # p_rank is the rank in policy, r_rank is the rank in rollout
+            communicator_index = create_nccl_comm(
+                nccl_group_id,
+                self.global_rank + command.src_replica_size,
+                self.world_size + command.src_replica_size,
+            )
+            # cache the communicator index
+            self.policy_to_rollout_nccl_communicators[nccl_unique_id_key] = (
+                communicator_index
+            )
+        if not hasattr(self, "policy_to_rollout_recv_insts"):
+            self.policy_to_rollout_recv_insts = []
+            try:
+                insts_meta = make_request_with_retry(
+                    partial(
+                        requests.post,
+                        json={
+                            "rank": self.global_rank,
+                        },
+                    ),
+                    self.get_alternative_urls(
+                        COSMOS_API_ROLLOUT_SHARD_RECV_INSTS_SUFFIX
+                    ),
+                    max_retries=constant.COSMOS_HTTP_RETRY_CONFIG.max_retries,
+                )
+                insts = msgpack.unpackb(insts_meta.content, strict_map_key=False)
+                self.policy_to_rollout_recv_insts = [
+                    WeightSyncInstructionsGroup.from_dict(inst) for inst in insts
+                ]
+            except Exception as e:
+                raise RuntimeError(
+                    f"[Rollout] Failed in fetching rollout from policy insts from controller after retries {e}."
+                )
+        with torch.cuda.stream(self.inference_stream):
+            # recv the weight from policy
+            st = time.time()
+            total_bytes_received = 0
+            for insts_group in self.policy_to_rollout_recv_insts:
+                # insts_group: WeightSyncInstructionsGroup -> inst collection for a full weight tensor
+                # handle inst group
+                total_bytes_received += self.recv_weight_shard(
+                    self.global_rank,
+                    insts_group,
+                    communicator_index,
+                    command.do_weight_sync_check,
+                )
+            time_eclapsed = time.time() - st
+            logger.debug(
+                f"[Rollout] All {len(self.policy_to_rollout_recv_insts)} at step {command.weight_step} recv operations finished in {time_eclapsed:.3f} seconds with {total_bytes_received / (1024 * 1024)} MB received."
+            )
+            self.cosmos_state.set_weight_synced()
+
+    @TRTLLMRolloutWorkerBase.register_rollout_command_handler(
+        RolloutToRolloutBroadcastCommand, backend="trtllm"
```

Comment:
For the instruction handling related function in this class, maybe we can merge with the vllm case which seems share similar steps? Which might be considered in later version.